packages/core/crates/core-path/src/channel_graph.rs

```diff
@@ -83,6 +106,40 @@ impl ChannelGraph {
     /// Returns a set of changes if the channel was already present in the graphs or
     /// None if the channel was not previously present in the channel graph.
     pub fn update_channel(&mut self, channel: ChannelEntry) -> Option<Vec<ChannelChange>> {
+        #[cfg(all(feature = "prometheus", not(test)))]
+        {
+            let (am_i_source, am_i_destination) = (&channel.source == &self.me, &channel.destination == &self.me);
+            if am_i_source || am_i_destination {
+                match channel.status {
+                    ChannelStatus::Closed => {
+                        if am_i_source {
```

Comment:
... and here match based on `channel.direction(self.me)`

Reply:
Even better, this allows for the `self.is_own_channel(...)` to be dropped completely, as it returns `None` for a channel not associated with me.